test/HatsProposalCreationWhitelistV1.test.ts

```diff
@@ -0,0 +1,309 @@
+
+      const whitelistedHats = await concreteHatsProposalCreationWhitelist.getWhitelistedHatIds();
+      expect(whitelistedHats.length).to.equal(1);
+      expect(whitelistedHats[0]).to.equal(proposerHatId2);
+    });
+
+    it('should emit HatRemovedFromWhitelist event when a hat is removed', async () => {
+      await expect(
+        concreteHatsProposalCreationWhitelist.connect(owner).removeHatFromWhitelist(proposerHatId1),
+      )
+        .to.emit(concreteHatsProposalCreationWhitelist, 'HatRemovedFromWhitelist')
+        .withArgs(proposerHatId1);
+    });
+
+    it('should not allow non-owner to remove a hat from the whitelist', async () => {
+      await expect(
+        concreteHatsProposalCreationWhitelist
+          .connect(nonOwner)
+          .removeHatFromWhitelist(proposerHatId1),
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+
+    it('should not allow removing a hat that is not whitelisted', async () => {
+      await expect(
+        concreteHatsProposalCreationWhitelist
+          .connect(owner)
+          .removeHatFromWhitelist(nonProposerHatId),
+      ).to.be.revertedWithCustomError(concreteHatsProposalCreationWhitelist, 'HatNotWhitelisted');
+    });
+  });
+
+  describe('isProposer override', () => {
+    runHatsProposerTests({
+      getMockHats: () => mockHats,
+      getContract: () => concreteHatsProposalCreationWhitelist,
+      hatWearer: () => hatWearer,
+      nonHatWearer: () => nonHatWearer,
+      tokenHolder: () => nonOwner, // Using nonOwner as tokenHolder since we don't have tokens in this test
+      owner: () => owner,
+      proposerHatId: proposerHatId1,
+      nonProposerHatId,
+    });
+  });
+
+  describe('getWhitelistedHatIds', () => {
+    it('should return the correct list of whitelisted hat IDs', async () => {
+      const whitelistedHats = await concreteHatsProposalCreationWhitelist.getWhitelistedHatIds();
+      expect(whitelistedHats.length).to.equal(2);
+      expect(whitelistedHats[0]).to.equal(proposerHatId1);
+      expect(whitelistedHats[1]).to.equal(proposerHatId2);
+    });
+
+    it('should return updated list after adding a hat', async () => {
+      await concreteHatsProposalCreationWhitelist.connect(owner).whitelistHat(nonProposerHatId);
+
+      const whitelistedHats = await concreteHatsProposalCreationWhitelist.getWhitelistedHatIds();
+      expect(whitelistedHats.length).to.equal(3);
+      expect(whitelistedHats[2]).to.equal(nonProposerHatId);
+    });
+
+    it('should return updated list after removing a hat', async () => {
+      await concreteHatsProposalCreationWhitelist
+        .connect(owner)
+        .removeHatFromWhitelist(proposerHatId1);
+
+      const whitelistedHats = await concreteHatsProposalCreationWhitelist.getWhitelistedHatIds();
+      expect(whitelistedHats.length).to.equal(1);
+      expect(whitelistedHats[0]).to.equal(proposerHatId2);
+    });
+  });
+
+  describe('ERC165', () => {
```

Comment:
As per the ERC-165 spec, no interface should ever match `0xffffffff`.

<img width="829" alt="image" src="https://github.com/user-attachments/assets/97e93416-05e6-44ba-9e0b-d111f8fbf626" />

Reply:
I can add a test for this